packages/relayer/processor/process_message.go

```diff
@@ -461,3 +452,52 @@ func (p *Processor) saveMessageStatusChangedEvent(
 
 	return nil
 }
+
+func (p *Processor) setGasTipOrPrice(ctx context.Context, auth *bind.TransactOpts) error {
+	gasTipCap, err := p.destEthClient.SuggestGasTipCap(ctx)
+	if err != nil {
+		if IsMaxPriorityFeePerGasNotFoundError(err) {
+			auth.GasTipCap = FallbackGasTipCap
+		} else {
+			gasPrice, err := p.destEthClient.SuggestGasPrice(context.Background())
+			if err != nil {
+				return errors.Wrap(err, "p.destBridge.SuggestGasPrice")
+			}
+			auth.GasPrice = gasPrice
+		}
+	}
+
+	auth.GasTipCap = gasTipCap
+
+	return nil
+}
+
+func (p *Processor) getCost(ctx context.Context, auth *bind.TransactOpts) (*big.Int, error) {
+	if auth.GasTipCap != nil {
+		blk, err := p.destEthClient.BlockByNumber(ctx, nil)
+		if err != nil {
+			return nil, err
+		}
+
+		var baseFee *big.Int
+
+		if p.taikoL2 != nil {
+			gasUsed := uint32(blk.GasUsed())
+			timeSince := uint64(time.Since(time.Unix(int64(blk.Time()), 0)))
```

Comment:
`blk.Time()` since `0`?

Reply:
This achieves time.Since() gives time.Now().Sub(y), this achieves time.Now() - blk.Time(). The 0 is the second param to time.Unix(), should work as intended!